src/mono/mono.proj

```diff
@@ -61,7 +61,7 @@
       <_MonoCXXFLAGS Include="-O0" />
       <_MonoCXXFLAGS Include="-ggdb3" />
       <_MonoCXXFLAGS Include="-fno-omit-frame-pointer" />
-      <_MonoCXXFLAGS Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
```

Comment:
This was introduced in https://github.com/dotnet/runtime/commit/7f2c158273104021c8e8b65dcbac1d623e132eae, and it looks like the motivation for this was that the libstdc++ ABI changed for some C++ stdlib constructs, like std::string, during the transition to C++11. The C++11 ABI is used by default in GCC 5 and above, but according to the aforementioned commit, we use CentOS 7 to build LLVM "nupkgs" (for... amd64? x86?) and that either uses GCC version < 5 or it explicitly sets `_GLIBCXX_USE_CXX11_ABI=0` when building LLVM.

So it's not really the case, if I'm understanding correctly, that LLVM-enabled mono will refuse to build on arm64--it's just that we're using the C++11 ABI on whatever machine we're using to build arm64 LLVM nupkgs (??), which isn't something that is inherently tied to arm64 as an architecture. It'd be helpful to know if we make (or should make) any hard guarantees about the minimum libstdc++ ABI we support for these LLVM nupkgs we are apparently building. And I think that in this case it'd be more sensible to make this preprocessor definition configurable via a flag or something.

Reply:
@imhameed I understand it's not arm specific but I didn't want to introduce a new global flag to set and made this quick workaround for me and Filip. I am not sure how to correctly handle this ABI difference..

Reply:
I don't have a strong opinion about this as a short-term workaround. Personally, I'd just add a flag to opt-in to the C++11 libstdc++ ABI, because dealing with this is a fact of life when linking dynamically to libstdc++, and I usually prefer to the oldest ABI available under the presumption that it will continue to work in the future. Maybe we should create an issue or something to discuss how we package pre-built LLVM binaries on Linux.

Plus this is still a problem on amd64: I have locally-built and patched copies of LLVM 6 and LLVM 9, but I built them with GCC 6 on Debian 9.12 ("stretch"), so my LLVM libraries dynamically link to libstdc++ using the C++11 ABI. mono.proj doesn't build out of the box for me with this setup, so as things are right now, I'd either have to manually delete this line or rebuild LLVM using `_GLIBCXX_USE_CXX11_ABI=0` to make progress.

Reply:
It depends on the build machine used to build LLVM and what options are used there.

By default the LLVM comes from NuGet packages built on Azure Pipelines (https://github.com/dotnet/llvm-project/blob/release/6.x/azure-pipelines.yml). The ABI version used there is currently implicitly set to the one of the machine (AFAIK). For Linux x64 that happens to be the old ABI (CentOS 7 packages). For Linux ARM/ARM64 it happens to be the new ABI (Ubuntu 16.04).

Alternatively, one can specify path to locally built LLVM and that may not match the definition here.

A reliable fix for both scenarios would probably be to force the LLVM builds to the old ABI explicitly on the lowest common denominator and keep the dotnet/runtime defines as-is. 

However, long term it's probably desirable to switch to the new ABI once CentOS 7 goes out of support. In fact, LLVM 10 will require that anyway. In that case it would make sense to keep the LLVM NuGets as-is (ie. default to the newest ABI of the build machine) and gradually remove this hack on the dotnet/runtime side. This PR goes towards this direction. For the currently supported architecture matrix it's probably okay to just exclude arm. Alternative would be to go the opposite way and restrict the condition to include x64 only.

Reply:
@imhameed It would make sense to add explicit override (to allow local builds with GCC 6+). That should be easy enough since top-level build script will pass any custom property here.

I'd personally prefer to restrict the usage of `-D_GLIBCXX_USE_CXX11_ABI=0` as much as possible. That would probably mean making it the default only when LLVM was obtained from NuGet and platform is x64.

Reply:
I restricted it to be defined only for 64 artifacts, also, introduced `MonoLLVMUseCxx11Abi` to be able to control it via `./build.sh`